test/functional/mempool_rebroadcast.py

```diff
@@ -0,0 +1,210 @@
+        self.log.info("Test simplest rebroadcast case")
+
+        node0 = self.nodes[0]
+        node1 = self.nodes[1]
+
+        # generate mempool transactions that both nodes know about
+        for _ in range(3):
+            node0.sendtoaddress(node1.getnewaddress(), 4)
+
+        self.sync_all()
+
+        # generate mempool transactions that only node0 knows about
+        disconnect_nodes(node0, 1)
+
+        for _ in range(3):
+            node0.sendtoaddress(node1.getnewaddress(), 5)
+
+        # check that mempools are different
+        assert_equal(len(node0.getrawmempool()), 6)
+        assert_equal(len(node1.getrawmempool()), 3)
+
+        # reconnect the nodes
+        connect_nodes(node0, 1)
+
+        # rebroadcast will only occur if there has been a block since the
+        # last run of CacheMinRebroadcastFee. when we connect a new peer, rebroadcast
+        # will be skipped on the first run, but caching will trigger.
+        # have node1 generate so there are still mempool txns that need to be synched.
+        node1.generate(1)
+
+        assert_equal(len(node1.getrawmempool()), 0)
+        wait_until(lambda: len(node0.getrawmempool()) == 3)
+
+        # bump time to hit rebroadcast interval
+        mocktime = int(time.time()) + 300 * 60
+        node0.setmocktime(mocktime)
+        node1.setmocktime(mocktime)
+
+        # check that node1 got txns bc rebroadcasting
+        wait_until(lambda: len(node1.getrawmempool()) == 3, timeout=30)
+
+    def test_rebroadcast_top_txns(self):
+        self.log.info("Testing that only txns with top fee rate get rebroadcast")
+
+        node = self.nodes[0]
+        node.setmocktime(0)
+
+        # mine a block to clear out the mempool
+        node.generate(1)
+        assert_equal(len(node.getrawmempool()), 0)
+
+        conn1 = node.add_p2p_connection(P2PTxInvStore())
+
+        # create txns
+        min_relay_fee = node.getnetworkinfo()["relayfee"]
+        txouts = gen_return_txouts()
+        utxo_count = 90
+        utxos = create_confirmed_utxos(min_relay_fee, node, utxo_count)
+        base_fee = min_relay_fee*100 # our transactions are smaller than 100kb
+        txids = []
+
+        # Create 3 batches of transactions at 3 different fee rate levels
+        range_size = utxo_count // 3
+
+        for i in range(3):
+            start_range = i * range_size
+            end_range = start_range + range_size
+            txids.append(create_lots_of_big_transactions(node, txouts, utxos[start_range:end_range], end_range - start_range, (i+1)*base_fee))
+
+        # 90 transactions should be created
+        # confirm the invs were sent (initial broadcast)
+        assert_equal(len(node.getrawmempool()), 90)
+        wait_until(lambda: len(conn1.tx_invs_received) == 90)
+
+        # confirm txns are more than max rebroadcast amount
+        assert_greater_than(node.getmempoolinfo()['bytes'], MAX_REBROADCAST_WEIGHT)
+
+        self.sync_all()
```

Comment:
I am unable to get this to pass in cadd572afd8b1468aee8a0a3ba65b5e38fc021dd or ffd154d06. Getting an 'Mempool sync timed out` AssertionError.

Reply:
should be fixed